src/Map/MapReader.cpp

```diff
@@ -34,8 +35,17 @@ Map Map::ReadMap(Stream::Reader& streamReader)
 	streamReader.Read<uint32_t>(map.tileInfos);
 	streamReader.Read<uint32_t>(map.terrainTypes);
 	ReadVersionTag(streamReader);
+
+	if (map.IsSavedGame()) {
+		ReadSavedGameSecondChunk(streamReader);
+	}
+
 	ReadVersionTag(streamReader);
-	ReadTileGroups(streamReader, map);
+	
+	if (!map.IsSavedGame()) {
+		ReadTileGroups(streamReader, map);
+	}
```

Comment:
I had never really thought about this before, but yeah, I guess the tile groups wouldn't be stored in saved game files.